### ISSUE TITLE
I'm having some troubles using enunciate + vaadin 

### ISSUE THREAD

**Issue by nenodias** [stoicflame/enunciate#87]
<h2>I'm having some troubles using enunciate + vaadin </h2>
<p>my-project is using resteasy for REST webservice and vaadin for other things.</p>

<pre>
[ERROR] Failed to execute goal org.codehaus.enunciate:maven-enunciate-plugin:1.30.1:assemble (default) on project my-project: Problem assembling the enunciate app. The invocation of APT has failed. See http://jira.codehaus.org/browse/ENUNCIATE-771 for a case where this has been seen. There is no explanation for this error. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.enunciate:maven-enunciate-plugin:1.30.1:assemble (default) on project my-project: Problem assembling the enunciate app.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:217)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Problem assembling the enunciate app.
	at org.codehaus.enunciate.AssembleMojo.execute(AssembleMojo.java:75)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: org.codehaus.enunciate.EnunciateException: The invocation of APT has failed. See http://jira.codehaus.org/browse/ENUNCIATE-771 for a case where this has been seen. There is no explanation for this error.
	at org.codehaus.enunciate.main.Enunciate.invokeApt(Enunciate.java:819)
	at org.codehaus.enunciate.main.Enunciate.doGenerate(Enunciate.java:401)
	at org.codehaus.enunciate.ConfigMojo$MavenSpecificEnunciate.doGenerate(ConfigMojo.java:670)
	at org.codehaus.enunciate.main.Enunciate$Stepper.step(Enunciate.java:1799)
	at org.codehaus.enunciate.main.Enunciate$Stepper.stepTo(Enunciate.java:1831)
	at org.codehaus.enunciate.AssembleMojo.execute(AssembleMojo.java:71)
	... 21 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
</pre>

<p>Thanks for attention</p>

**Comment by stoicflame** [comment]
These kinds of errors are almost always compile errors.

Try running a compile without Enunciate to see if there are any compile errors.

```
mvn -Denunciate.skip=true compile
```

**Comment by nenodias** [comment]
Skip the enunciate compile works...
But I stay without doc...

:/

**Comment by stoicflame** [comment]
What happens when you try it with a Javac check?

```
mvn -Denunciate.javac.check=true clean install
```

**Comment by nenodias** [comment]
This's the return. Broke on layout class of my Vaadin app
Has a way to skip vaadin classes and their packages?
<pre>
Downloading: https://oss.sonatype.org/content/repositories/springsource-milestones/org/jvnet/staxex/stax-ex/maven-metadata.xml
[WARNING] Could not transfer metadata org.jvnet.staxex:stax-ex/maven-metadata.xml from/to springsource-milestones (https://oss.sonatype.org/content/repositories/springsource-milestones): Failed to transfer file: https://oss.sonatype.org/content/repositories/springsource-milestones/org/jvnet/staxex/stax-ex/maven-metadata.xml. Return code is: 503 , ReasonPhrase:Service Unavailable.
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ myproject-supervisor ---
[INFO] Deleting /opt/arquivos/projects/myproject-supervisor/target
[INFO] 
[INFO] --- maven-enunciate-plugin:1.30.1:assemble (default) @ myproject-supervisor ---
Note: /opt/arquivos/projects/myproject-supervisor/src/main/java/br/com/myproject/supervisor/ui/layout/ManutencaoMenuWebTreeView.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
[INFO] initializing enunciate.
[INFO] [csharp] C# compilation is disabled, but the source code will still be generated.
[INFO] invoking enunciate:generate step...

warning: The apt tool and its associated API are planned to be
removed in the next major JDK release.  These features have been
superseded by javac and the standardized annotation processing API,
javax.annotation.processing and javax.lang.model.  Users are
recommended to migrate to the annotation processing features of
javac; see the javac man page for more information.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.701s
[INFO] Finished at: Thu Jun 11 13:44:56 BRT 2015
[INFO] Final Memory: 49M/348M
[INFO] ------------------------------------------------------------------------
</pre>

**Comment by stoicflame** [comment]
> Broke on layout class of my Vaadin app

I'm not seeing that in the output you pasted above. Am I missing something?

> Has a way to skip vaadin classes and their packages?

Yes, see [Excluding Classes](https://github.com/stoicflame/enunciate/wiki/Excluding-Classes).

**Comment by nenodias** [comment]
Hi I'm try change my enunciate.xml
<pre>
&lt;?xml version=&quot;1.0&quot;?&gt;
&lt;enunciate label=&quot;full&quot; xmlns:xsi=&quot;http://www.w3.org/2001/XMLSchema-instance&quot;
           xsi:noNamespaceSchemaLocation=&quot;http://enunciate.webcohesion.com/schemas/enunciate-1.30.xsd&quot; excludeUnreferencedClasses=&quot;true&quot; includeReferencedClasses=&quot;true&quot;&gt;

	&lt;api-classes&gt;
	 	&lt;include pattern=&quot;br.com.myproject.supervisor.ws.**&quot;/&gt;
	 &lt;/api-classes&gt;

	&lt;webapp mergeWebXML=&quot;src/main/webapp/WEB-INF/web.xml&quot; /&gt;
	
	&lt;modules&gt;
	 	&lt;csharp disableCompile=&quot;true&quot; disabled=&quot;true&quot;&gt;&lt;/csharp&gt;
	 	&lt;c disabled=&quot;true&quot;&gt;&lt;/c&gt;
		&lt;jersey  useSubcontext=&quot;false&quot; &gt;&lt;/jersey&gt;
		&lt;docs docsDir=&quot;/api&quot; splashPackage=&quot;br.com.myproject&quot; title=&quot;MyProject Service API&quot; &gt;&lt;/docs&gt;
	&lt;/modules&gt;
	
&lt;/enunciate&gt;
</pre>

I try only include my Rest classes, but still broke
Out of console is:
<pre>
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building supervisor 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading: https://oss.sonatype.org/content/repositories/springsource-milestones/org/jvnet/staxex/stax-ex/maven-metadata.xml
[WARNING] Could not transfer metadata org.jvnet.staxex:stax-ex/maven-metadata.xml from/to springsource-milestones (https://oss.sonatype.org/content/repositories/springsource-milestones): Failed to transfer file: https://oss.sonatype.org/content/repositories/springsource-milestones/org/jvnet/staxex/stax-ex/maven-metadata.xml. Return code is: 503 , ReasonPhrase:Service Unavailable.
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ sinc-supervisor ---
[INFO] Deleting /opt/arquivos/projects/myproject-supervisor/target
[INFO] 
[INFO] --- maven-enunciate-plugin:1.30:assemble (default) @ sinc-supervisor ---
Note: /opt/arquivos/projects/myproject-supervisor/src/main/java/br/com/myproject/supervisor/ui/layout/ManutencaoMenuWebTreeView.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
[INFO] initializing enunciate.
[INFO] invoking enunciate:generate step...

warning: The apt tool and its associated API are planned to be
removed in the next major JDK release.  These features have been
superseded by javac and the standardized annotation processing API,
javax.annotation.processing and javax.lang.model.  Users are
recommended to migrate to the annotation processing features of
javac; see the javac man page for more information.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.945s
[INFO] Finished at: Fri Jun 12 11:22:36 BRT 2015
[INFO] Final Memory: 43M/345M
[INFO] ------------------------------------------------------------------------
</pre>
Say about my ManutencaoMenuWebTreeView.java, but is don't included to enunciate.
The plugin configuration:
<pre>
&lt;plugin&gt;
				&lt;groupId&gt;org.codehaus.enunciate&lt;/groupId&gt;
				&lt;artifactId&gt;maven-enunciate-plugin&lt;/artifactId&gt;
				&lt;version&gt;${enunciate.version}&lt;/version&gt;
				&lt;configuration&gt;
					&lt;configFile&gt;enunciate.xml&lt;/configFile&gt;
					&lt;javacSourceVersion&gt;${java.version}&lt;/javacSourceVersion&gt;
					&lt;javacTargetVersion&gt;${java.version}&lt;/javacTargetVersion&gt;
					&lt;javacCheck&gt;true&lt;/javacCheck&gt;
				&lt;/configuration&gt;
				&lt;executions&gt;
					&lt;execution&gt;
						&lt;goals&gt;
							&lt;goal&gt;assemble&lt;/goal&gt;
						&lt;/goals&gt;
					&lt;/execution&gt;
				&lt;/executions&gt;
	      &lt;/plugin&gt;
</pre>

**Comment by stoicflame** [comment]
I think `ManutencaoMenuWebTreeView.java` is a red herring; I don't think that's the real problem.

We need to isolate the problem spot. You may need to start removing code until it works, then add stuff back in until it fails.

**Comment by stoicflame** [comment]
Any progress on this?

**Comment by nenodias** [comment]
Does not work excludes the classes on enunciate.xml.
The project compile works without enunciate plugin.
Thanks for attention, but I'm not know what to do for fix it

**Comment by stoicflame** [comment]
At this point, you've got two options:

1. Start taking out code until it works, then add back in code piece by piece until it breaks in order to isolate the piece that breaks it.
2. Figure out a way for me to reproduce the issue on my end so I can identify whether there's a bug.

**Comment by nenodias** [comment]
Hi, please you can try fix on my Test App
https://github.com/nenodias/sistema-vaadin

If I uncomment the executions of the enunciate plugin not his generate the documentation.

**Comment by stoicflame** [comment]
Just wanted to give an update. I've cloned your repo, but I can't build at the moment because of [SourceForge's outage](http://sourceforge.net/blog/sourceforge-infrastructure-and-service-restoration/), but I hope to try it again soon after they're back up.

Specifically, the request for [this Maven artifact](http://jasperreports.sourceforge.net/maven2/com/lowagie/itext/2.1.7.js1/) needs to _not_ return 503.

**Comment by nenodias** [comment]
Itext it's used by ireport, try remove it from pom

**Comment by stoicflame** [comment]
Error confirmed.

I've spent some time digging into it, but the problem seems to be deep within the APT tool. As implied by the error message, I've spent some time on an issue like this before, but never could get to the bottom of it.

However, since Enunciate 2.0 will not be using APT any more, we may just need to wait until 2.0 is ready.

**Comment by nenodias** [comment]
Okay, thanks for reply
^^

**Comment by stoicflame** [comment]
I'm going to claim this fixed in Enunciate 2.0.0.
